src/NLog/Logger.cs

```diff
@@ -403,10 +407,57 @@ private LogEventInfo PrepareLogEventInfo(LogEventInfo logEvent)
             {
                 logEvent.FormatProvider = this.Factory.DefaultCultureInfo;
             }
+
+            HandleMessageTemplate(logEvent);
+
             return logEvent;
 
         }
 
+        /// <summary>
+        /// Parse and render message template.
+        /// </summary>
+        /// <param name="logEvent"></param>
+        private static void HandleMessageTemplate(LogEventInfo logEvent)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)